### PR TITLE
[NCL-8180] Allow to provide configuration when requesting generation

### PR DESCRIPTION
**Pull request by goldmann**
This change makes it possible to provide the Config with the product and
generator configuration as part of the REST API call to
/api/v1alpha1/sboms/generate/build/{buildId}. It enables providing a
custom configuration for that particular generation.

In case the configuration is provided it will be used, which means that
the initialziation phase which includes crafrting configuration won't be
run at all.

This feature is very useful for cases where we want to try out different
configurations to "get it right", for example when onboarding someone.
Another use case is to adjust configuration so that SBOM can be
generated for an old build.

Configuration can be sent in the body as YAML (application/yaml
content type) or JSON (application/json content type). Provided
configuration is validated and adjusted if necessary (putting default
values, if not provided).

### CODE REVIEW

**Comment by goldmann** [comment]
Will mark this as a draft, will extend it with tests.

File: core/src/main/java/org/jboss/sbomer/core/config/ConfigSchemaValidator.java [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+        try {
+            InputStream is = SchemaValidator.class.getClassLoader().getResourceAsStream("schemas/config.json");
+            schema = new String(is.readAllBytes(), StandardCharsets.UTF_8);
+        } catch (IOException e) {
+            throw new ApplicationException("Could not read the configuration file schema", e);
+        }
+
+        try {
+            String writeValueAsString = ObjectMapperProvider.json().writeValueAsString(config);
+            System.out.println(writeValueAsString);
```
Comment (vibe13) [inline core/src/main/java/org/jboss/sbomer/core/config/ConfigSchemaValidator.java:62]: Leftover or needed?

**Review by vibe13 — APPROVED** [review]
@goldmann great work, which will be handy to test out very easily new builds! Thanks!

File: core/src/main/java/org/jboss/sbomer/core/config/ConfigSchemaValidator.java (hunk 1 above)
Comment (goldmann) [reply core/src/main/java/org/jboss/sbomer/core/config/ConfigSchemaValidator.java:62]: Leftover!

Comment (goldmann) [reply core/src/main/java/org/jboss/sbomer/core/config/ConfigSchemaValidator.java]: Removed!

**Review by vibe13 — APPROVED** [review]

**Comment by aloubyansky** [comment]
Is this feature documented somewhere?
In case of passing custom config, would the resulting manifest be persisted?
Is there an option to generate a config w/o storing it permanently?

**Comment by goldmann** [comment]
1. I'll give you the link shortly.
2. Yes.
3. No.

**Pull request merged**